src/Main.ts

```diff
@@ -321,10 +327,19 @@ export class Main {
         return this.getGhostForSlack(message.user_id, teamDomain, teamId);
     }
 
-    public async getGhostForSlack(slackUserId: string, teamDomain: string, teamId?: string): Promise<SlackGhost> {
+    public async getGhostForSlack(slackUserId: string, teamDomain?: string, teamId?: string): Promise<SlackGhost> {
+        let domain: string;
```

Comment:
it feels weird to need to use another variable here, but I couldn't find a way without assertion...

```
if (!teamDomain && !teamId) {
            throw Error("Must provide either a teamDomain or a teamId");
        }
        if (! teamDomain) {
            teamDomain = await this.getTeamDomainForMessage({team_id: teamId});
        }
        const userId = this.getUserId(
            slackUserId.toUpperCase(),
            <string>teamDomain,
        );
```

feels nicer and needs no extra var, but needs an assertion - any way around this?

Reply:
Shortened it a bit, but we need the extra var because `teamDomain` is `string|undefined`